pkg/beholder/httpclient.go

```diff
@@ -0,0 +1,203 @@
+	}
+	loggerProvider := sdklog.NewLoggerProvider(
+		sdklog.WithResource(loggerResource),
+		sdklog.WithProcessor(loggerProcessor),
+	)
+	logger := loggerProvider.Logger(defaultPackageName)
+
+	// Tracer
+	tracerProvider, err := newHTTPTracerProvider(cfg, baseResource, tlsConfig)
+	if err != nil {
+		return noop, err
+	}
+	tracer := tracerProvider.Tracer(defaultPackageName)
+
+	// Meter
+	meterProvider, err := newHTTPMeterProvider(cfg, baseResource, tlsConfig)
+	if err != nil {
+		return noop, err
+	}
+	meter := meterProvider.Meter(defaultPackageName)
+
+	// Message Emitter
+	var messageLogProcessor sdklog.Processor
+	if cfg.EmitterBatchProcessor {
+		messageLogProcessor = sdklog.NewBatchProcessor(
+			sharedLogExporter,
+			sdklog.WithExportTimeout(cfg.EmitterExportTimeout), // Default is 30s
+		)
+	} else {
+		messageLogProcessor = sdklog.NewSimpleProcessor(sharedLogExporter)
+	}
+
+	messageAttributes := []attribute.KeyValue{
+		attribute.String("beholder_data_type", "custom_message"),
+	}
+	messageLoggerResource, err := sdkresource.Merge(
+		sdkresource.NewSchemaless(messageAttributes...),
+		baseResource,
+	)
+	if err != nil {
+		return noop, err
+	}
+
+	messageLoggerProvider := sdklog.NewLoggerProvider(
+		sdklog.WithResource(messageLoggerResource),
+		sdklog.WithProcessor(messageLogProcessor),
+	)
+	messageLogger := messageLoggerProvider.Logger(defaultPackageName)
+
+	emitter := messageEmitter{
+		messageLogger: messageLogger,
+	}
+
+	onClose := func() (err error) {
+		for _, provider := range []shutdowner{messageLoggerProvider, loggerProvider, tracerProvider, meterProvider, messageLoggerProvider} {
+			err = errors.Join(err, provider.Shutdown(context.Background()))
+		}
+		return
+	}
+	return &Client{cfg, logger, tracer, meter, emitter, loggerProvider, tracerProvider, meterProvider, messageLoggerProvider, onClose}, nil
+}
+
+func newHTTPTracerProvider(config Config, resource *sdkresource.Resource, tlsConfig *tls.Config) (*sdktrace.TracerProvider, error) {
+	ctx := context.Background()
+
+	tlsConfigOption := otlptracehttp.WithInsecure()
+	if tlsConfig != nil {
+		tlsConfigOption = otlptracehttp.WithTLSClientConfig(tlsConfig)
+	}
+	exporter, err := otlptracehttp.New(ctx,
+		tlsConfigOption,
+		otlptracehttp.WithEndpoint(config.OtelExporterHTTPEndpoint),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	opts := []sdktrace.TracerProviderOption{
+		sdktrace.WithBatcher(exporter, trace.WithBatchTimeout(config.TraceBatchTimeout)), // Default is 5s
+		sdktrace.WithResource(resource),
+		sdktrace.WithSampler(
+			sdktrace.ParentBased(
+				sdktrace.TraceIDRatioBased(config.TraceSampleRatio),
+			),
+		),
+	}
+	if config.TraceSpanExporter != nil {
+		opts = append(opts, sdktrace.WithBatcher(config.TraceSpanExporter))
+	}
+	return sdktrace.NewTracerProvider(opts...), nil
+}
+
+func newHTTPMeterProvider(config Config, resource *sdkresource.Resource, tlsConfig *tls.Config) (*sdkmetric.MeterProvider, error) {
+	ctx := context.Background()
```

Comment:
Are these "unused"?

Reply:
Addressed https://github.com/smartcontractkit/chainlink-common/pull/830/commits/90367b568661d9822e17544193b2f1d4ac88a2a4

Reply:
In otlpmetrichttp.New ctx is not used, checking others

Reply:
OK, maybe worth noting in a comment :+1:

Reply:
otlptracehttp.New does use context

Reply:
otlploghttp.New does not use context

Reply:
> OK, maybe worth noting in a comment 👍

added comments